pkg/cache/cache_with_eviction.go

```diff
@@ -57,7 +57,10 @@ func NewLRUWithEviction[K comparable, V any](reg prometheus.Registerer, maxEntri
 
 // onEvicted is called when an entry is evicted from the underlying LRU.
 func (c *LRUWithEviction[K, V]) onEvicted(k K, v V) {
-	go c.onEvictedCallback(k, v)
```

Comment:
Let's remove the goroutine separately. It needs more testing and thought. There's already a semaphore to create a backpressure around this. I don't believe this is the root cause. It's there to ensure we don't pay the cost of eviction while adding items.

Reply:
Good call! I completely missed that there's a semaphore, great stuff

Reply:
See comment below